homeassistant/components/litterrobot/switch.py

```diff
@@ -0,0 +1,89 @@
+
+    entities = []
+    for robot in hub.account.robots:
+        entities.append(LitterRobotSwitch(robot, NIGHT_LIGHT, hub))
+        entities.append(LitterRobotSwitch(robot, PANEL_LOCKOUT, hub))
+        entities.append(LitterRobotSwitch(robot, SLEEP_MODE, hub))
+
+    if entities:
+        async_add_entities(entities, True)
+
+
+class LitterRobotSwitch(LitterRobotEntity, ToggleEntity):
+    """Litter-Robot Switches."""
+
+    @property
+    def is_on(self):
+        """Return true if switch is on."""
+        if self.entity_type == NIGHT_LIGHT:
+            return self.robot.night_light_active
+        if self.entity_type == PANEL_LOCKOUT:
+            return self.robot.panel_lock_active
+        if self.entity_type == SLEEP_MODE:
+            return self.robot.sleep_mode_active
+
+    @property
+    def icon(self):
+        """Return the icon based on the entity_type."""
+        if self.entity_type == NIGHT_LIGHT:
+            return "mdi:lightbulb-on" if self.is_on else "mdi:lightbulb-off"
+        if self.entity_type == PANEL_LOCKOUT:
+            return "mdi:lock" if self.is_on else "mdi:lock-open"
+        if self.entity_type == SLEEP_MODE:
+            return "mdi:sleep" if self.is_on else "mdi:sleep-off"
+
+    async def async_turn_on(self, **kwargs):
+        """Turn the switch on."""
+        if self.entity_type == NIGHT_LIGHT:
+            await self.perform_action_and_refresh(self.robot.set_night_light, True)
+        elif self.entity_type == PANEL_LOCKOUT:
+            await self.perform_action_and_refresh(self.robot.set_panel_lockout, True)
+        elif self.entity_type == SLEEP_MODE:
+            await self.perform_action_and_refresh(
+                self.robot.set_sleep_mode,
+                True,
+                self.parse_time_at_default_timezone(DEFAULT_SLEEP_TIME),
+            )
+
+    async def async_turn_off(self, **kwargs):
+        """Turn the switch off."""
+        if self.entity_type == NIGHT_LIGHT:
+            await self.perform_action_and_refresh(self.robot.set_night_light, False)
+        elif self.entity_type == PANEL_LOCKOUT:
+            await self.perform_action_and_refresh(self.robot.set_panel_lockout, False)
+        elif self.entity_type == SLEEP_MODE:
+            await self.perform_action_and_refresh(self.robot.set_sleep_mode, False)
+
+    @property
+    def device_state_attributes(self):
+        """Return device specific state attributes."""
+        if self.entity_type == SLEEP_MODE:
+            [start_time, end_time] = [None, None]
+            if self.is_on:
+                start_time = dt_util.as_local(
+                    self.robot.sleep_mode_start_time
+                ).strftime("%H:%M:00")
+                end_time = dt_util.as_local(self.robot.sleep_mode_end_time).strftime(
+                    "%H:%M:00"
+                )
+            return {
+                "start_time": start_time,
+                "end_time": end_time,
+            }
```

Comment:
All datetimes need to be in `isoformat()`

Reply:
@bdraco, start_time and end_time are recurring times, meaning they repeat every day at the same time. This is much more similar to how HA handles an input_datetime that has only a time component. So is an isoformat still necessary on this?

Reply:
I'm not sure. Let's ask for a second opinion on this.

Reply:
I have one of these, and Nate is correct. They have no concept of day-of-the-week for sleep mode; they only accept a time normalized to the local timezone (which is being done here in code).

First, I looked at the implementation of input_datetime, and it recognizes when someone puts in just a time... and then the entity "expands" to the next day when the time is reached when the input_datetime component fires. 

For an internal entity, this is probably appropriate; however, in this case, the external entity (the LitterRobot) doesn't have a need nor a purpose for the day at all. Storing the isoformat() time just means that we have to waste time internally updating the attribute state to add the date each day when the time is reached and then turn around and strftime() the time away from the isoformat() time again. Instead, we just set it and don't look at it again until the app config changes it or an event updates it... which leads me to...

Second, it's pretty uncommon on this device to continually twiddle the time. Once sleep time is set (that's the time when the litterbox does not cycle), it's usually not touched again for a simple reason: it'd freak them out if it moves at times they don't expect. Cats don't have day-of-the-week support either. :)

Personally, I'd give this one a pass.

Reply:
> "Cats don't have day-of-the-week support either."

That made me chuckle. But precisely what you said about updating the attribute state, which is why I went the route of showing only a time piece.